src/int64_tensors.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {Shape, util} from '@tensorflow/tfjs';
+import {endianness} from 'os';
+
+const INT32_MAX = 2147483648;
+
+/**
+ * Node.js-specific tensor type: int64-type scalar.
+ *
+ * This class is created for a specific purpose: to support
+ * writing `step`s to TensorBoard via op-kernel bindings.
+ * `step` is required to have an int64 dtype, but TensorFlow.js
+ * (tfjs-core) doesn't have a built-in int64 dtype. This is
+ * related to a lack of `Int64Array` or `Uint64Array` typed
+ * array in basic JavaScript.
+ *
+ * This class is introduced as a workaround.
+ */
+export class Int64Scalar {
```

Comment:
I'm a little confused with this class still.

It represents an `int64` but JS can't handle those types today. It tries to parse the high/low parts of the number and validate endianness? This is mostly confusing because I think we just wanted to check that the numbers are within `INT32_MIN` and `INT32_MAX`. Why do we need all the various other stuff - seems somewhat expensive for not doing anything.